### ISSUE TITLE
devtools::load_all files not included in deployment suggestion

### ISSUE THREAD

**Issue by aronatkins** [rstudio/rstudio#6090]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.767
    OS Version      : macOS Catalina 10.15.2 (19C57) 
    R Version       : R version 3.6.2 (2019-12-12)

### Steps to reproduce the problem

Create a new project. In that project, create three files:

The primary document, `index.Rmd`:

````markdown
---
title: using devtools::load_all
---

```{r}
devtools::load_all("bananas")
```

```{r}
bananas()
```
````

A `bananas/DESCRIPTION` file:

```
Package: stuff
Version: 0.0.1
```

A `bananas/R/bananas.R` file:

```r
bananas <- function() { cat("bananas!\n") }
```

This file renders successfully and includes the "bananas!" text in its output.

Attempt to deploy this content with source to RStudio Connect. Only the original `index.Rmd` is suggested; the internal package loaded with `devtools::load_all` is missed.

### Describe the problem in detail

![image](https://user-images.githubusercontent.com/362187/73073110-9f73e800-3e84-11ea-8d27-c1f70b6c4b7d.png)

### Describe the behavior you expected

The entire `bananas` directory should be included as a deployment suggestion.

### Workaround

Manually include the contents of the `bananas` directory. This is a little tedious, as there does not appear to be a way to indicate that everything beneath a directory should be included (macOS).

![image](https://user-images.githubusercontent.com/362187/73073569-9df6ef80-3e85-11ea-97cc-f741b4013557.png)

**Comment by aronatkins** [comment]
CC @andrie

**Comment by andrie** [comment]
Note that `devtools::load_all()` will generally also expect a `DESCRIPTION` file.

**Comment by aronatkins** [comment]
@andrie Yep. My example included an Rmd, a DESCRIPTION, and an R file within the "package".

**Comment by jmcphers** [comment]
What is the motivating/real-world example for this? The fix would not be in the IDE proper but would need to be done with some combination of coordinating changes in rmarkdown (where we discover needed files) and packrat (where we discover needed packages).

**Comment by aronatkins** [comment]
Motivating example:

While developing an R Markdown document, some folks want to have supporting R code alongside that document. That R code will eventually shift to a package, but it is not formally a package. The `load_all` function lets that code be handled mostly as you would a package. The loaded code can also be contained within a namespace rather than as globally exported objects.

**Comment by andrie** [comment]
I *think* some of this has been resolved, but commenting here to prevent the issue from getting stale.  I'd be happy to contribute more information and/or report this on other projects, if appropriate.
